exporters/otlp/otlphttp/options_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlphttp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type env map[string]string
+
+func (e *env) getEnv(env string) string {
+	return (*e)[env]
+}
+
+func TestWithBackoff(t *testing.T) {
+	tests := []struct {
+		name    string
+		opts    []Option
+		env     env
+		asserts func(t *testing.T, c *config)
+	}{
+		{
+			name: "Test default configs",
+			asserts: func(t *testing.T, c *config) {
+				assert.Equal(t, "localhost:4317", c.traces.endpoint)
+				assert.Equal(t, "localhost:4317", c.metrics.endpoint)
+				assert.Equal(t, NoCompression, c.traces.compression)
+				assert.Equal(t, NoCompression, c.metrics.compression)
+			},
+		},
+		{
+			name: "Test With Endpoint",
+			opts: []Option{
+				WithEndpoint("someendpoint"),
+			},
+			asserts: func(t *testing.T, c *config) {
+				assert.Equal(t, "someendpoint", c.traces.endpoint)
+				assert.Equal(t, "someendpoint", c.metrics.endpoint)
+			},
+		},
+		{
+			name: "Test With Endpoint",
+			opts: []Option{
+				WithTracesEndpoint("trace_endpoint"),
+				WithMetricsEndpoint("metrics_endpoint"),
+			},
+			asserts: func(t *testing.T, c *config) {
+				assert.Equal(t, "trace_endpoint", c.traces.endpoint)
+				assert.Equal(t, "metrics_endpoint", c.metrics.endpoint)
+			},
+		},
```

Comment:
Missing test for environment setting of the endpoint(s) and tests to check the environment vs code specified configuration precedent.

Reply:
I've added 5 tests for each configuration, them being: `With` configs, `With` signal specific configs, Env configs, Env specific configs and mixed Env and `With` configs.

Reply:
Awesome, looks good :+1: